post_query_generators.php

```diff
@@ -0,0 +1,383 @@
+                $survey_choice_id . ", " . $course_survey_choice_id . ", " . $ta_survey_choice_id . ", '" . $name . "', " . $term . ", '" .
+                $default_survey_open . "', '" . $default_survey_close . "');";
+             break;
+        case "course":
+            return "INSERT INTO surveys (dept_survey_choices, course_survey_choices, ta_survey_choices, name, term, default_survey_open, default_survey_close) VALUES (" .
+                $dept_survey_choice_id . ", " . $survey_choice_id . ", " . $ta_survey_choice_id . ", '" . $name . "', " . $term . ", '" .
+                $default_survey_open . "', '" . $default_survey_close . "');";
+            break;
+        case "section":
+            return "INSERT INTO surveys (dept_survey_choices, course_survey_choices, ta_survey_choices, name, term, default_survey_open, default_survey_close) VALUES (" .
+                $dept_survey_choice_id . ", " . $course_survey_choice_id . ", " . $survey_choice_id . ", '" . $name . "', " . $term . ", '" .
+                $default_survey_open . "', '" . $default_survey_close . "');";
+            break,
+    }
+}
+
+/**
+ * This function is for returning an SQL statement which will get the choice_id back
+ * from the "survey_choice" table given the survey_choice_id
+ * @param survey_choice_id:int The id of dept/course/ta_survey_choices
+ * @param level:string dept/course/section
+ * @return string: An SQL statement
+ */
+function query_get_choice_id (
+    $survey_choice_id,
+    $level
+) {
+    switch ($level) {
+        case 'dept':
+            return "SELECT choices_id FROM dept_survey_choices WHERE id = " .
+                $survey_choice_id . ";";
+            break;
+
+        case 'course':
+            return "SELECT choices_id FROM course_survey_choices WHERE id = " .
+                $survey_choice_id . ";";
+            break;
+
+        default:
+            return "SELECT choices_id FROM ta_survey_choices WHERE id = " .
+                $survey_choice_id . ";";
+            break;
+    }
+}
+
+/**
+ * This function is for returning an SQL statement which update the choice instance
+ * @param choices_id:int The choice instance that the user wants to update
+ * @param level:string dept/course/section
+ * @param dept_survey_choices:object contains the new choice object
+ * @param course_survey_choices:object contains the new choice object
+ * @param ta_survey_choices: object contains the new choice obejct
+ */
+function query_update_choices (
+    $choices_id,
+    $level,
+    $dept_survey_choices,
+    $course_survey_choices,
+    $ta_survey_choices
+) {
+    switch ($level) {
+        case "dept":
+            return "UPDATE choices ON choice1 = " .
+                $dept_survey_choices['choices'][0] . ", " .
+                "choice2 = " . $dept_survey_choices['choices'][1] . ", " .
```

Comment:
never construct SQL queries this way. They are vulnerable to injection attacks. Look at `get_info.php` and assemble your queries that way.

Reply:
I see.